pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -128,345 +167,552 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			return ctrl.Result{}, nil
 		}
 
-		log.Error(err, errGetES)
+		log.Error(err, logErrorGetES)
 		syncCallsError.With(resourceLabels).Inc()
-
 		return ctrl.Result{}, err
 	}
 
-	// See https://github.com/external-secrets/external-secrets/issues/3604
-	// We fetch the ExternalSecret resource above, however the status subresource is inconsistent.
-	// We have to explicitly fetch it, otherwise it may be missing and will cause
-	// unexpected side effects.
-	err = r.SubResource("status").Get(ctx, &externalSecret, &externalSecret)
-	if err != nil {
-		log.Error(err, "failed to get status subresource")
-		return ctrl.Result{}, err
-	}
-
-	timeSinceLastRefresh := 0 * time.Second
-	if !externalSecret.Status.RefreshTime.IsZero() {
-		timeSinceLastRefresh = time.Since(externalSecret.Status.RefreshTime.Time)
-	}
-
 	// skip reconciliation if deletion timestamp is set on external secret
-	if externalSecret.DeletionTimestamp != nil {
-		log.Info("skipping as it is in deletion")
+	if !externalSecret.GetDeletionTimestamp().IsZero() {
+		log.V(1).Info("skipping ExternalSecret, it is marked for deletion")
 		return ctrl.Result{}, nil
 	}
 
 	// if extended metrics is enabled, refine the time series vector
 	resourceLabels = ctrlmetrics.RefineLabels(resourceLabels, externalSecret.Labels)
 
+	// skip this ExternalSecret if it uses a ClusterSecretStore and the feature is disabled
 	if shouldSkipClusterSecretStore(r, externalSecret) {
-		log.Info("skipping cluster secret store as it is disabled")
+		log.V(1).Info("skipping ExternalSecret, ClusterSecretStore feature is disabled")
```

Comment:
So this guy should set

```go
	r.markAsDone(externalSecret, start, log, esv1beta1.ConditionReasonSecretSynced, msgSynced)

```

To update the timestamp on the latest reconciled event.

Reply:
I strongly disagree as this violates the idea of controller classes.

Specifically, this check indicates that this controller should completely be ignoring this ExternalSecret, with the expectation that another instance is managing it.

Therefore, we should not be updating the status of this ExternalSecret.

Reply:
If you want to change this behavior, I recommend we discuss what the intention of disabling `ClusterSecretStore` is, probably in a separate PR.

If we don't want to account for the multiple controller case, the correct state would also be "failed" with some condition explaining that we can't reconcile this ExternalSecret because it uses a ClusterSecretStore.